src/Symfony/Component/Messenger/Envelope.php

```diff
@@ -48,22 +48,26 @@ public function with(StampInterface ...$stamps): self
         $cloned = clone $this;
 
         foreach ($stamps as $stamp) {
-            $cloned->stamps[\get_class($stamp)] = $stamp;
+            $cloned->stamps[\get_class($stamp)][] = $stamp;
         }
 
         return $cloned;
     }
 
-    public function get(string $stampFqcn): ?StampInterface
+    public function last(string $stampFqcn): ?StampInterface
     {
-        return $this->stamps[$stampFqcn] ?? null;
+        return isset($this->stamps[$stampFqcn]) ? end($this->stamps[$stampFqcn]) : null;
     }
 
     /**
-     * @return StampInterface[] indexed by fqcn
+     * @return StampInterface[]|StampInterface[][] The stamps for the specified FQCN, or all stamps by their class name
```

Comment:
Would it be better to have 2 different methods instead of two kind of returned value?

Reply:
not sure personally: when reading, `all()` or `all(SomeStampp:class)` is very clear to me. Discoverability looks better also (same as `EventDispatcher::getListeners()` btw) + types-wise, the interface has no method so autocompletion won't help.

Reply:
I think you're both right. 
Having one method is fine, but isn't grouping them by FQCN in one case a relic from how we used to manage stamps of the same type before ? 
If now stamps should also give us an history of the life of the message, it would feel more natural if `all()` (without argument) did just return a `StampInterface[]` with all the stamps in the order they were added (ie: not grouped by FQCN anymore).